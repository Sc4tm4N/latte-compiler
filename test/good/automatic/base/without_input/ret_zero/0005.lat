// print int variables according to int max (compared to C)
int main() {
  int INT_MAX = 2147483647;
  printInt(INT_MAX + INT_MAX);
  printInt(INT_MAX - INT_MAX);
  printInt(INT_MAX + 2 * INT_MAX);
  printInt(INT_MAX - 2 * INT_MAX);
  printInt(7 * INT_MAX);
  printInt(INT_MAX * 7);
  printInt(INT_MAX / 7);
  printInt(7 / INT_MAX);
  printInt(7 % INT_MAX);
  printInt(INT_MAX % 7);
  printInt(INT_MAX * INT_MAX);
  printInt(INT_MAX * INT_MAX * (INT_MAX * INT_MAX) / INT_MAX - INT_MAX * INT_MAX);
  printInt(INT_MAX * INT_MAX * (INT_MAX * INT_MAX) / INT_MAX + INT_MAX * INT_MAX + 5);
  return 0;
}