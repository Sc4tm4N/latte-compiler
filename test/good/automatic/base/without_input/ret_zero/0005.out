-2
0
2147483645
-2147483647
2147483641
2147483641
306783378
0
7
1
1
-1
6
